packages/arb-token-bridge-ui/src/state/app/utils.ts

```diff
@@ -47,7 +52,43 @@ export const getDepositStatus = (tx: Transaction | MergedTransaction) => {
     return DepositStatus.L1_PENDING
   }
 
-  // l1 succeeded...
+  // for teleport txn
+  if (
+    isTeleport({
+      sourceChainId: tx.parentChainId, // we make sourceChain=parentChain assumption coz it's a deposit txn
+      destinationChainId: tx.childChainId
+    }) &&
+    tx.l1ToL2MsgData &&
+    tx.l2ToL3MsgData
```

Comment:
If for any reason `tx.l2ToL3MsgData` is undefined and it is teleport tx, it will go to the next step and possibly get wrong status in the `l1ToL2MsgData` switch.

I think it's safer to check for `tx.l1ToL2MsgData` and `tx.l2ToL3MsgData` inside the teleport block here.